module/batch-action/src/Domain/ValueObject/BatchActionAction.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\BatchAction\Domain\ValueObject;
+
+class BatchActionAction
+{
+    public const MAX_LENGTH = 20;
+
+    private string $value;
+
+    public function __construct(string $value)
+    {
+        $this->value = strtolower(trim($value));
+
+        if (!self::isValid($this->value)) {
+            throw new \InvalidArgumentException(\sprintf('Value "%s" is not valid batch action', $value));
+        }
+    }
+
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+    public function __toString(): string
+    {
+        return $this->value;
+    }
+
+    public function isEqual(self $code): bool
+    {
+        return $code->value === $this->value;
+    }
+
+    public static function isValid(string $value): bool
+    {
+        return '' !== $value && \strlen($value) <= self::MAX_LENGTH;
```

Comment:
I think we will also need "real" value validation to do available actions. Not now maybe but worth considering

Reply:
I do not understand what you mean by "real"

Reply:
we may have only some actions available for specific resource. I.e `delete` for product only :)

Reply:
We cannot predict what the values will be depending on the modules installed, so such a VO has to do the validation from external information (e.g. from a database), and this can be a bit difficult directly in this object.

Reply:
Agreed ;) It was just a suggestion for the future, not an expectation for now ;)